### ISSUE TITLE
Windows runner agents are killed at 72 hours with the default runner installation

### ISSUE THREAD

**Issue by dawilk** [github-aws-runners/terraform-aws-github-runner#4066]
I set up a multi-runner config to provide runners of various operating systems. One of those multi-runner configs is for windows runners. Because new runners can take 10-20 minutes to provision, I set up an idle config that keeps at least one runner always online.

Here's the issue: the default start-runner script [uses Task Scheduler to handle the runner task](https://github.com/philips-labs/terraform-aws-github-runner/blob/7b911b2ea7ef70c99236e77e2122ef9833d327cc/modules/runners/templates/start-runner.ps1#L128), and the default behavior of tasks like this is to timeout/stop if they are running for 72hrs. If the runner stays online, this results in a dead/offline runner until the task on that instance is re-triggered. This also prevents scale-up of new runners, completely blocking the runner group from use.

I think there are a few fixes for this:
- Set the task ExecutionTimeLimit to `PT0S` (never), allowing the task to run indefinitely. It'll still result in a dead runner if it stops/crashes for any reason.
- Add more triggers to the task to make sure it starts again if the task ends. Self-healing, but still possible to have periods of unreliable runners.
- Switch to Windows Service instead of Task Scheduler by adding the `--runasservice` flag to [the config command](https://github.com/philips-labs/terraform-aws-github-runner/blob/7b911b2ea7ef70c99236e77e2122ef9833d327cc/modules/runners/templates/start-runner.ps1#L110). Was there a design decision for using Task Scheduler over Windows Service? This seems like it would be the best long-term fix.

In any case, the current start-runner.ps1 results in very fragile Windows runners. It looks like the only way to avoid it without forking the module is to replace the _entire_ user_data because the [start-runner.ps1 is a hard-coded local](https://github.com/philips-labs/terraform-aws-github-runner/blob/7b911b2ea7ef70c99236e77e2122ef9833d327cc/modules/runners/main.tf#L58).

**Comment by dawilk** [comment]
> To me the option to switch as `--runasservice` seems the most simple and reliable one. Does this option conflicts with ephemeral runners?

No, the runasservice flag just tells the agent to wrap itself with a Service instead of a run-once Task. Ephemeral runners would be terminating the whole instance after the run, so whether it's a Task or a Service starting the agent will not matter.

I'm not sure if I'll be able to come back and implement/test this change. I ended up updating my idle_config to recycle the runners after business hours the same way I do other OSes. I'll eventually be generating custom AMIs to speed up the launch time, so the OOB start-runner and idle_config will not matter.

**Comment by npalm** [comment]
Thanks, in case you have any change that you can share, either via an issue or a PR that improve the module. That is always welcome.
